packages/workflow/src/utils.ts

```diff
@@ -1,37 +1,41 @@
-/* eslint-disable @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-return */
-
-type Serializable = { toJSON?: () => string };
-
-export const deepCopy = <T>(source: T, hash = new WeakMap(), path = ''): T => {
+/* eslint-disable @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-argument */
+type Primitives = string | number | boolean | bigint | symbol | null | undefined;
+export const deepCopy = <T extends ((object | Date) & { toJSON?: () => string }) | Primitives>(
+	source: T,
+	hash = new WeakMap(),
+	path = '',
+): T => {
 	let clone: any;
+	let i: any;
```

Comment:
what's the advantage of having an explicitly `any` variable that's used in multiple contexts?

Reply:
it is for performance reasons, it's cheaper to reuse an existing variable then to declare a new

Reply:
is the performance difference large enough to justify the loss of typings and readability?

Reply:
just ran some perf test and there are no significant differences so let's go with better typing